pkg/commands/venv.go

```diff
@@ -1,16 +1,168 @@
 package commands
 
-// GetShell detects which shell to run, and returns the command to run it.
-func GetShell(osLookupEnv func(key string) (string, bool)) string {
-	shell, ok := osLookupEnv("OKCTL_SHELL")
-	if ok {
-		return shell
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/mishudark/errors"
+	"github.com/oslokommune/okctl/pkg/binaries/run/awsiamauthenticator"
+	"github.com/oslokommune/okctl/pkg/binaries/run/kubectl"
+	"github.com/oslokommune/okctl/pkg/config"
+	"github.com/oslokommune/okctl/pkg/helm"
+	"github.com/oslokommune/okctl/pkg/okctl"
+)
+
+// OkctlEnvironment contains data about the okctl environment
+type OkctlEnvironment struct {
+	Region                 string
+	AWSAccountID           string
+	Environment            string
+	Repository             string
+	ClusterName            string
+	UserDataDir            string
+	Debug                  bool
+	KubectlBinaryDir       string
+	AwsIamAuthenticatorDir string
+}
+
+// Validate the inputs
+func (o *OkctlEnvironment) Validate() error {
+	return validation.ValidateStruct(o,
+		validation.Field(&o.Environment, validation.Required),
+		validation.Field(&o.AWSAccountID, validation.Required),
+		validation.Field(&o.Region, validation.Required),
+		validation.Field(&o.ClusterName, validation.Required),
+		validation.Field(&o.UserDataDir, validation.Required),
+		validation.Field(&o.KubectlBinaryDir, validation.Required),
+		validation.Field(&o.AwsIamAuthenticatorDir, validation.Required),
+	)
+}
+
+// GetOkctlEnvironment returns data needed to connect to an okctl cluster
+func GetOkctlEnvironment(o *okctl.Okctl) (OkctlEnvironment, error) {
+	meta := o.RepoStateWithEnv.GetMetadata()
+	cluster := o.RepoStateWithEnv.GetCluster()
+
+	userDataDir, err := o.GetUserDataDir()
+	if err != nil {
+		return OkctlEnvironment{}, err
+	}
+
+	k, err := o.BinariesProvider.Kubectl(kubectl.Version)
+	if err != nil {
+		return OkctlEnvironment{}, err
+	}
+
+	a, err := o.BinariesProvider.AwsIamAuthenticator(awsiamauthenticator.Version)
+	if err != nil {
+		return OkctlEnvironment{}, err
+	}
+
+	opts := OkctlEnvironment{
+		Region:                 meta.Region,
+		AWSAccountID:           cluster.AWSAccountID,
+		Environment:            cluster.Environment,
+		Repository:             meta.Name,
+		ClusterName:            cluster.Name,
+		UserDataDir:            userDataDir,
+		Debug:                  o.Debug,
+		KubectlBinaryDir:       path.Dir(k.BinaryPath),
+		AwsIamAuthenticatorDir: path.Dir(a.BinaryPath),
+	}
+
+	err = opts.Validate()
+	if err != nil {
+		return OkctlEnvironment{}, errors.E(err, "failed to validate show credentials options")
```

Comment:
This function is used in both `show credentials` and `venv` isn't it? This error message might prove to be confusing if it triggers in `venv`

Reply:
Fixed, added new error msg